assets/js/services/metadata.js

```diff
@@ -362,34 +395,74 @@ export function prepRelatedUrl(items = []) {
  * Remove helper labels from Batch Edit form post data
  * @param {Object} batchAdds
  * @param {Object} batchDeletes
+ * @param {Object} batchReplaces
  * @param {Boolean} hasAdds
  * @param {Boolean} hasDeletes
- *
+ * @param {Boolean} hasReplaces
  * @returns {Object}
  */
 export function removeLabelsFromBatchEditPostData(
   batchAdds,
   batchDeletes,
+  batchReplaces,
   hasAdds,
-  hasDeletes
+  hasDeletes,
+  hasReplaces
 ) {
-  let returnObj = { add: { descriptiveMetadata: {} }, delete: {} };
+  let returnObj = {
+    add: { descriptiveMetadata: {}, administrativeMetadata: {} },
+    delete: {},
+    replace: { administrativeMetadata: {}, descriptiveMetadata: {} },
+  };
 
   if (hasAdds) {
-    Object.keys(batchAdds.descriptiveMetadata).forEach((key) => {
-      returnObj.add.descriptiveMetadata[key] = batchAdds.descriptiveMetadata[
-        key
-      ].map((item) => {
-        // Regular string value
-        if (typeof item !== "object") {
-          return item;
+    batchAdds.descriptiveMetadata &&
+      Object.keys(batchAdds.descriptiveMetadata).forEach((key) => {
+        returnObj.add.descriptiveMetadata[key] = batchAdds.descriptiveMetadata[
+          key
+        ].map((item) => {
```

Comment:
@divyareddynu Looks like we're re-using this code block (lines 423 - 432) three separate times.  Just to make our life easier in the future to maintain this code, how about we pull this out into it's own helper function?